pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -150,6 +154,16 @@ func (grm *goRoutineMap) operationComplete(
 	}
 }
 
+func (grm *goRoutineMap) IsOperationPending(operationName string) bool {
+	grm.lock.Lock()
```

Comment:
Since you are only reading from the data structure, get a readonly lock (`RLock`).